pyproject.toml

```diff
@@ -64,6 +64,7 @@ ignore = [
     "Q002",
     "Q003",
     "W191",
+    "A002" # Ignore 'id' attribute shadowing built-in 'id'
```

Comment:
We should where we can use `id_` as that is the pythonic way to deal with collisions (see [PEP-8](https://peps.python.org/pep-0008/#:~:text=single_trailing_underscore_%3A%20used%20by%20convention%20to%20avoid%20conflicts%20with%20Python%20keyword%2C%20e.g.)).

Further, a global deactivation of argument shadowing is bad. If we really need it, it should be on a file-by-file basis but would suggest to try using id_ where possible.